applicationset/generators/generator_spec_processor.go

```diff
@@ -81,3 +91,30 @@ func mergeGeneratorTemplate(g Generator, requestedGenerator *argoprojiov1alpha1.
 
 	return *dest, err
 }
+
+// Currently for Matrix Generator. Allows interpolating the matrix's 2nd child generator with values from the 1st child generator
+// "params" parameter is an array, where each index corresponds to a generator. Each index contains a map w/ that generator's parameters.
+// Since the Matrix generator currently only allows 2 child generators, effectively this params array will always be size 1 (containing the 1st child generator's params)
+// Uses similar process for interpreting values as the actual Application Template
+func interpolateGenerator(requestedGenerator *argoprojiov1alpha1.ApplicationSetGenerator, params []map[string]string) error {
```

Comment:
@KojoRising like @Lobstrosity said returning generator from interpolation would solve ur problem of resusbstituting as original generator will not be replaced like this

```
func interpolateGenerator(requestedGenerator *argoprojiov1alpha1.ApplicationSetGenerator, params []map[string]string) (argoprojiov1alpha1.ApplicationSetGenerator, error) {
	newrequestedGenerator := *requestedGenerator
	tmplBytes, err := json.Marshal(newrequestedGenerator)
	if err != nil {
		log.WithError(err).WithField("requestedGenerator", requestedGenerator).Error("error marshalling requested generator for interpolation")
		return newrequestedGenerator, err
	}

	render := utils.Render{}
	fstTmpl := fasttemplate.New(string(tmplBytes), "{{", "}}")
	replacedTmplStr, err := render.Replace(fstTmpl, params[0], true)
	if err != nil {
		log.WithError(err).WithField("interpolatedGeneratorString", replacedTmplStr).Error("error interpolating generator with other generator's parameter")
		return newrequestedGenerator, err
	}

	err = json.Unmarshal([]byte(replacedTmplStr), requestedGenerator)
	if err != nil {
		log.WithError(err).WithField("requestedGenerator", requestedGenerator).Error("error unmarshalling requested generator for interpolation")
		return newrequestedGenerator, err
	}
	return newrequestedGenerator, nil
}
```